File: LeetCode/cpp/1720_decode_XORed_array.cc
/*
Difficulty: Easy
*/

#include <vector>

class Solution {
 public:
  std::vector<int> decode(const std::vector<int>& encoded, const int first) {
    std::vector<int> original;
    original.reserve(encoded.size() + 1);
    original.push_back(first);

    int currDecoded = first;
    for (const int codedValue : encoded) {
      currDecoded ^= codedValue;
      original.push_back(currDecoded);
    }

    return original;
  }
};

File: LeetCode/rust/1720_decode_XORed_array.rs
/*
Difficulty: Easy
*/

impl Solution {
    pub fn decode(encoded: Vec<i32>, first: i32) -> Vec<i32> {
        let mut original = Vec::with_capacity(encoded.len() + 1);
        original.push(first);

        let mut curr_original = first;
        for encoded_value in encoded {
            curr_original ^= encoded_value;
            original.push(curr_original);
        }

        original
    }
}
